React/Modules/RCTHapticFeedback.m

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "RCTHapticFeedback.h"
+#import <UIKit/UIKit.h>
+
+@implementation RCTHapticFeedback
+
+{
+  UIImpactFeedbackGenerator *_impactFeedback;
+  UINotificationFeedbackGenerator *_notificationFeedback;
+  UISelectionFeedbackGenerator *_selectionFeedback;
+}
+@synthesize bridge = _bridge;
+
+RCT_EXPORT_MODULE()
+
+- (void)setBridge:(RCTBridge *)bridge
+{
+  _bridge = bridge;
+  if ([UIFeedbackGenerator class]) {
+    _impactFeedback = [UIImpactFeedbackGenerator new];
```

Comment:
`UIImpactFeedbackGenerator` has `initWithStyle` which has 3 styles of vibrations.